src/Microsoft.FeatureManagement.Telemetry.ApplicationInsights.AspNetCore/TargetingTelemetryInitializer.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+//
+
+using Microsoft.ApplicationInsights.AspNetCore.TelemetryInitializers;
+using Microsoft.ApplicationInsights.Channel;
+using Microsoft.ApplicationInsights.DataContracts;
+using Microsoft.AspNetCore.Http;
+
+namespace Microsoft.FeatureManagement.Telemetry.ApplicationInsights.AspNetCore
+{
+    /// <summary>
+    /// Used to add targeting information to outgoing Application Insights telemetry.
+    /// </summary>
+    public class TargetingTelemetryInitializer : TelemetryInitializerBase
+    {
+        /// <summary>
+        /// Creates an instance of the TargetingTelemetryInitializer
+        /// </summary>
+        public TargetingTelemetryInitializer(IHttpContextAccessor httpContextAccessor) : base(httpContextAccessor)
+        {
+        }
+
+        /// <summary>
+        /// When telemetry is initialized, adds targeting information to all relevant telemetry.
+        /// </summary>
+        /// <param name="httpContext">The <see cref="HttpContext"/> to get the targeting information from.</param>
+        /// <param name="requestTelemetry">The <see cref="RequestTelemetry"/> relevant to the telemetry.</param>
+        /// <param name="telemetry">The <see cref="ITelemetry"/> to be initialized.</param>
+        /// <exception cref="ArgumentNullException">Thrown if the any param is null.</exception>
+        protected override void OnInitializeTelemetry(HttpContext httpContext, RequestTelemetry requestTelemetry, ITelemetry telemetry)
+        {
+            if (telemetry == null)
+            {
+                throw new ArgumentNullException("telemetry");
+            }
+
+            if (requestTelemetry == null)
+            {
+                throw new ArgumentNullException("requestTelemetry");
+            }
+
+            if (httpContext == null)
+            {
+                throw new ArgumentNullException("httpContext");
+            }
+
+            string targetingId = (string) httpContext.Items["TargetingId"];
```

Comment:
httpContext.Items["TargetingId"] is expected to be set as a string only by the TargetingHttpContextMiddleware. It seems like we cannot prevent other middleware from modifying it. I suggest we add some error handling here in case that httpContext.Items["TargetingId"] is not a string.

Reply:
Great point. I went ahead and adjusted it to `?.ToString();`, which doesn't have error handling necessarily, but it'll either get a string result or null from whatever is placed in there. This seems right to me, what do you think?